delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/organizations/OrganizationsImpl.scala

```diff
@@ -0,0 +1,257 @@
+    uuidF()
+      .flatMap { uuid =>
+        eval(CreateOrganization(label, uuid, description, caller))
+      }
+      .named("createOrganization", component)
+
+  override def update(
+      label: Label,
+      description: Option[String],
+      rev: Long
+  )(implicit caller: Subject): IO[OrganizationRejection, OrganizationResource] =
+    eval(UpdateOrganization(label, rev, description, caller)).named("updateOrganization", component)
+
+  override def deprecate(
+      label: Label,
+      rev: Long
+  )(implicit caller: Subject): IO[OrganizationRejection, OrganizationResource] =
+    eval(DeprecateOrganization(label, rev, caller)).named("deprecateOrganization", component)
+
+  override def fetch(label: Label): UIO[Option[OrganizationResource]] =
+    agg.state(label.value).map(_.toResource).named("fetchOrganization", component)
+
+  override def fetchAt(label: Label, rev: Long): IO[RevisionNotFound, Option[OrganizationResource]] =
+    if (rev == 0L) UIO.pure(None).named("fetchOrganizationAt", component)
+    else {
+      eventLog
+        .currentEventsByPersistenceId(s"$entityType-$label", Long.MinValue, Long.MaxValue)
+        .takeWhile(_.event.rev <= rev)
+        .fold[OrganizationState](OrganizationState.Initial) {
+          case (state, event) => Organizations.next(state, event.event)
+        }
+        .compile
+        .last
+        .hideErrors
+        .flatMap {
+          case Some(state) if state.rev == rev => UIO.pure(state.toResource)
+          case Some(_)                         =>
+            fetch(label).flatMap {
+              case Some(res) => IO.raiseError(RevisionNotFound(rev, res.rev))
+              case None      => IO.pure(None)
+            }
+          case None                            => IO.raiseError(RevisionNotFound(rev, 0L))
+        }
+        .named("fetchOrganizationAt", component)
+    }
+
+  override def fetch(uuid: UUID): UIO[Option[OrganizationResource]] =
+    fetchFromCache(uuid)
+      .flatMap {
+        case Some(label) => fetch(label)
+        case None        => UIO.pure(None)
+      }
+      .named("fetchOrganizationByUuid", component)
+
+  override def fetchAt(uuid: UUID, rev: Long): IO[RevisionNotFound, Option[OrganizationResource]] =
+    fetchFromCache(uuid)
+      .flatMap {
+        case Some(label) => fetchAt(label, rev)
+        case None        => UIO.pure(None)
+      }
+      .named("fetchOrganizationAtByUuid", component)
+
+  private def fetchFromCache(uuid: UUID): UIO[Option[Label]]                                  =
+    cache.collectFirst { case (label, resource) if resource.value.uuid == uuid => label }
+
+  private def eval(cmd: OrganizationCommand): IO[OrganizationRejection, OrganizationResource] =
+    for {
+      evaluationResult <- agg.evaluate(cmd.label.value, cmd).mapError(_.value)
+      resource         <- IO.fromOption(evaluationResult.state.toResource, UnexpectedInitialState(cmd.label))
+      _                <- cache.put(cmd.label, resource)
```

Comment:
I think that maybe we should ignore errors for the cache update, as it would be recovered by the async index, so something like: `cache.put(cmd.label, resource).as(()).recoverWith(_ => IO.unit)`

Reply:
cache.put returns an `UIO`. There is nothing to recover from it.